src/lib/rules/jsll-script-included/jsll-script-included.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @fileoverview This rule confirms that JSLL script is included in the page
+ */
+import { Category } from '../../enums/category';
+import { RuleContext } from '../../rule-context';
+import { IAsyncHTMLDocument, IAsyncHTMLElement, IRule, IRuleBuilder, ITraverseEnd, Severity } from '../../types';
+import { isHTMLDocument } from '../../utils/misc';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        const validate = async (data: ITraverseEnd) => {
+            const { resource }: { resource: string } = data;
+            const jsllDir = `https://az725175.vo.msecnd.net/scripts/jsll-`;
+            const passRegex = new RegExp(`^(\\d+\\.)js`);
+            const warningRegex = new RegExp(`^(\\d+\\.){2,}js`);
+
+            // Messages.
+            const noScriptInHeadMsg = `No JSLL script was included in the <head> tag.`;
+            const redundantScriptInHeadMsg = `More than one JSLL scripts were included in the <head> tag.`;
+            const warningScriptVersionMsg = `Use the latest release of JSLL with 'jsll-4.js'. It is not recommended to specify the version number unless you wish to lock to a specific release.`;
+            const invalidScriptVersionMsg = `The jsll script versioning is not valid.`;
+            const wrongScriptOrderMsg = `The JSLL script isn't placed prior to other scripts.`;
+
+            const getJsllScripts= (elements: Array<IAsyncHTMLElement>) => {
+                return elements.filter((element) => {
+                    return element.getAttribute('src').startsWith(jsllDir);
+                });
+            };
+
+            if (!isHTMLDocument(resource, context.pageHeaders)) {
+                return;
+            }
+
+            const pageDOM: IAsyncHTMLDocument = context.pageDOM as IAsyncHTMLDocument;
+            const scriptsInHead: Array<IAsyncHTMLElement> = await pageDOM.querySelectorAll('head script');
+            const jsllScriptsInHead: Array<IAsyncHTMLElement> = getJsllScripts(scriptsInHead);
+
+            if (!jsllScriptsInHead.length) {
+                await context.report(resource, null, noScriptInHeadMsg);
+
+                return;
+            }
+
+            if (jsllScriptsInHead.length > 1) {
+                await context.report(resource, null, redundantScriptInHeadMsg);
```

Comment:
Maybe flag each script that shouldn't be included? That way it's easier to know exactly which scripts should be removed.